.github/workflows/ci-build-image.yml

```diff
@@ -129,3 +141,40 @@ jobs:
     - uses: skjolber/maven-cache-github-action@v1
       with:
         step: save
+
+  retest-integration-tests:
+    runs-on: ubuntu-latest
+    if: ${{ !inputs.skip-itests && inputs.run-tests-only }}
```

Comment:
Maybe this should also have `&& inputs.pr-number != ''` or something like that? Otherwise it seems like this will always fail on pulling the image later on.

Reply:
Same for the `sha-value`.

This would really just be to safeguard against doing something silly in the future and trying to reuse this workflow without passing the inputs that it actually needs, since the inputs here are scoped to the whole file rather than this individual job.

I suppose that's also an argument to split this up into a separate reusable workflow so that the inputs can be properly defined as required for this particular job...

Reply:
will try the `&& inputs.pr-number != ''` and same for `sha-value` for now and I will look into separating the building image and integrated testing into separate workflows next week or if i have time on the weekend